wave_kinematics/src/quadrotor.cpp

```diff
@@ -0,0 +1,435 @@
+    this->states(2) = 0.0;   // yaw
+    this->states(3) = 0.0;   // angular velocity x
+    this->states(4) = 0.0;   // angular velocity y
+    this->states(5) = 0.0;   // angular velocity z
+    this->states(6) = 0.0;   // x
+    this->states(7) = 0.0;   // y
+    this->states(8) = 0.0;   // z
+    this->states(9) = 0.0;   // linear velocity x
+    this->states(10) = 0.0;  // linear velocity y
+    this->states(11) = 0.0;  // linear velocity z
+
+    this->Ix = 0.0963;  // inertial x
+    this->Iy = 0.0963;  // inertial y
+    this->Iz = 0.1927;  // inertial z
+
+    this->kr = 0.1;  // rotation drag constant
+    this->kt = 0.2;  // translation drag constant
+
+    this->l = 0.9;  // arm length
+    this->d = 1.0;  // drag
+
+    this->m = 1.0;   // mass of quad
+    this->g = 10.0;  // gravitational constant
+
+    this->attitude_setpoints = Vec4();
+    this->position_setpoints = VecX(3);
+
+    this->attitude_controller = AttitudeController();
+    this->position_controller = PositionController();
+}
+
+QuadrotorModel::QuadrotorModel(VecX pose) {
+    this->states = VecX(12);
+    this->states(0) = pose(3);  // roll
+    this->states(1) = pose(4);  // pitch
+    this->states(2) = pose(5);  // yaw
+    this->states(3) = 0.0;      // angular velocity x
+    this->states(4) = 0.0;      // angular velocity y
+    this->states(5) = 0.0;      // angular velocity z
+    this->states(6) = pose(0);  // x
+    this->states(7) = pose(1);  // y
+    this->states(8) = pose(2);  // z
+    this->states(9) = 0.0;      // linear velocity x
+    this->states(10) = 0.0;     // linear velocity y
+    this->states(11) = 0.0;     // linear velocity z
+
+    this->Ix = 0.0963;  // inertial x
+    this->Iy = 0.0963;  // inertial y
+    this->Iz = 0.1927;  // inertial z
+
+    this->kr = 0.1;  // rotation drag constant
+    this->kt = 0.2;  // translation drag constant;
+
+    this->l = 0.9;  // arm length
+    this->d = 1.0;  // drag
+
+    this->m = 1.0;   // mass of quad
+    this->g = 10.0;  // gravitational constant
+
+    this->attitude_setpoints = Vec4();
+    this->attitude_setpoints << 0.0, 0.0, 0.0, 0.5;
+
+    this->position_setpoints = VecX(3);
+    this->position_setpoints << pose(0), pose(1), pose(2);
+
+    this->attitude_controller = AttitudeController();
+    this->position_controller = PositionController();
+}
+
+int QuadrotorModel::update(VecX motor_inputs, double dt) {
+    double ph, th, ps;
+    double p, q, r;
+    double x, y, z;
+    double vx, vy, vz;
+
+    double Ix, Iy, Iz;
+    double kr, kt;
+    double tauf, taup, tauq, taur;
+    double m, g;
+
+    Mat4 A;
+    Vec4 tau;
+
+    // setup
+    ph = this->states(0);
+    th = this->states(1);
+    ps = this->states(2);
+
+    p = this->states(3);
+    q = this->states(4);
+    r = this->states(5);
+
+    x = this->states(6);
+    y = this->states(7);
+    z = this->states(8);
+
+    vx = this->states(9);
+    vy = this->states(10);
+    vz = this->states(11);
+
+    Ix = this->Ix;
```

Comment:
It looks like these assignments are just to get around the extra length of `this->` huh?
Maybe change the rules to something convenient, instead of going to great lengths to circumvent them ;-) - -same for clang-format directives on long lines.

Or, to be clear about what you're doing
```cpp
const auto &Ix = this->Ix`;
```

This also works to keep calculations readable while having meaningful,  non-abbreviated member names.